include/robot_state_publisher/joint_state_listener.h

```diff
@@ -37,24 +37,29 @@
 #ifndef JOINT_STATE_LISTENER_H
 #define JOINT_STATE_LISTENER_H
 
+#include <map>
+#include <string>
+
+#include <boost/scoped_ptr.hpp>
+
 #include <urdf/model.h>
 #include <kdl/tree.hpp>
 #include <ros/ros.h>
 #include <sensor_msgs/JointState.h>
 
 #include "robot_state_publisher/robot_state_publisher.h"
 
-using namespace std;
-using namespace ros;
-using namespace KDL;
-
 typedef boost::shared_ptr<sensor_msgs::JointState const> JointStateConstPtr;
 typedef std::map<std::string, urdf::JointMimicSharedPtr > MimicMap;
 
 namespace robot_state_publisher {
 
 class JointStateListener {
 public:
+  /** Default constructor.
+   */
+  JointStateListener();
```

Comment:
I haven't looked through the member types, but does JointStateListener meet the requirements to get a default copy or move assignment operator? If not, then I don't see a way to go from a default constructed `JointStateListener` to a functional one. Mind adding a unit test to check if the assignment operator works?

Reply:
Made sure `JointStateListener` move assignment works in bc0cb46